pkg/app/log.go

```diff
@@ -45,33 +45,34 @@ func DefineLoggingFlags(cmd *kingpin.CmdClause) {
 }
 
 // SetupLogging sets logger formatter and level.
-func SetupLogging(runtimeConfig *config.Config) {
-	jsonFormatter := log.JSONFormatter{DisableTimestamp: LogNoTime}
-	textFormatter := log.TextFormatter{DisableTimestamp: LogNoTime, DisableColors: true}
-	colorFormatter := log.TextFormatter{DisableTimestamp: LogNoTime, ForceColors: true, FullTimestamp: true}
-	switch strings.ToLower(LogType) {
-	case "json":
-		log.SetFormatter(&jsonFormatter)
-	case "text":
-		log.SetFormatter(&textFormatter)
-	case "color":
-		log.SetFormatter(&colorFormatter)
-	default:
-		log.SetFormatter(&jsonFormatter)
-	}
-	if LogProxyHookJSON {
-		formatter := log.StandardLogger().Formatter
-		log.SetFormatter(&ProxyJsonWrapperFormatter{WrappedFormatter: formatter})
-	}
+func SetupLogging(runtimeConfig *config.Config, logger *unilogger.Logger) {
+	// TODO: if we need formatters - add to logger
+	// jsonFormatter := log.JSONFormatter{DisableTimestamp: LogNoTime}
+	// textFormatter := log.TextFormatter{DisableTimestamp: LogNoTime, DisableColors: true}
+	// colorFormatter := log.TextFormatter{DisableTimestamp: LogNoTime, ForceColors: true, FullTimestamp: true}
+	// switch strings.ToLower(LogType) {
+	// case "json":
+	// 	log.SetFormatter(&jsonFormatter)
+	// case "text":
+	// 	log.SetFormatter(&textFormatter)
+	// case "color":
+	// 	log.SetFormatter(&colorFormatter)
+	// default:
+	// 	log.SetFormatter(&jsonFormatter)
+	// }
+	// if LogProxyHookJSON {
+	// 	formatter := log.StandardLogger().Formatter
+	// 	log.SetFormatter(&ProxyJsonWrapperFormatter{WrappedFormatter: formatter})
+	// }
 
-	setLogLevel(LogLevel)
+	unilogger.SetDefaultLevel(unilogger.LogLevelFromStr(LogLevel))
 
 	runtimeConfig.Register("log.level",
 		fmt.Sprintf("Global log level. Default duration for debug level is %s", ForcedDurationForDebugLevel),
 		strings.ToLower(LogLevel),
 		func(oldValue string, newValue string) error {
 			log.Infof("Set log level to '%s'", newValue)
-			setLogLevel(newValue)
+			unilogger.SetDefaultLevel(unilogger.LogLevelFromStr(LogLevel))
```

Comment:
If I understand correctly, the level should be set via newValue?

Reply:
we start working with logger here https://github.com/flant/shell-operator/blob/feature/slog/cmd/shell-operator/main.go#L40 and make it default. 
with this command we set global logger, which is a local logger and all global calls

Reply:
it's really interesting place for discuss. think we can refactor this later.

Reply:
sry fixed